app/TaskRouter/WorkspaceFacade.php

```diff
@@ -3,104 +3,182 @@
 namespace App\TaskRouter;
 
 
+use Twilio\Exceptions\DeserializeException;
+use Twilio\Exceptions\TwilioException;
+
 class WorkspaceFacade
 {
-    private $taskRouterClient;
+    private $_taskRouterClient;
+
+    private $_workspace;
 
-    private $workspace;
+    private $_activities;
+
+    public function __construct($taskRouterClient, $workspace)
+    {
+        $this->_taskRouterClient = $taskRouterClient;
+        $this->_workspace = $workspace;
+    }
 
-    public function __construct($taskRouterClient, $params)
+    public static function createNewWorkspace($taskRouterClient, $params)
     {
-        $this->taskRouterClient = $taskRouterClient;
         $workspaceName = $params["friendlyName"];
         foreach ($taskRouterClient->workspaces()->read() as $workspace) {
```

Comment:
You can pass a friendlyName parameter to the read method and it will filter workspaces